### PR TITLE
Add bosses tag

### PR DESCRIPTION
**Pull request by MichaelTRoss0**
Add EntityType tag for bosses.

By default, the tag contains Ender Dragon and Wither.

This PR is a prerequisite for https://github.com/MinecraftForge/MinecraftForge/pull/8442

### CODE REVIEW

**Comment by MichaelTRoss0** [comment]
Generated data is coming very soon

**Comment by MichaelTRoss0** [comment]
This PR is a prerequisite for https://github.com/MinecraftForge/MinecraftForge/pull/8442

File: src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+package net.minecraftforge.common.data;
+
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.tags.EntityTypeTagsProvider;
+import net.minecraft.world.entity.EntityType;
+import net.minecraftforge.common.Tags;
+
+public class ForgeEntityTypeTagsProvider extends EntityTypeTagsProvider {
```
Comment (sciwhiz12) [inline src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java]: Formatting: braces on newlines. Additionally, to conform with its sibling tags providers, please mark this class as `final`.
```suggestion
public final class ForgeEntityTypeTagsProvider extends EntityTypeTagsProvider
{
```

File: src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
```
Comment (sciwhiz12) [inline src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java]: Outdated license header, please update this PR to the latest commit of the base branch and run the `updateLicenses`/`licenseFormat` task.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Please update this PR to the latest commit of the 1.18.2 branch.

File: src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java (hunk 2 above)
Comment (MichaelTRoss0) [reply src/main/java/net/minecraftforge/common/data/ForgeEntityTypeTagsProvider.java]: I'm a bit late, but I have made the required changes.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]
Your PR has outdated license headers; please update your PR to the latest commit of the base branch (1.18.x), then run the `updateLicenses`/`licenseFormat` task.

**Comment by sciwhiz12** [comment]
Hello! @MichaelTRoss0, please leave a comment whether you wish to continue this PR and update it to the latest active version of 1.19.2 (or leave an explanation why this PR's changes are not applicable to 1.19.2). If we do not hear back from you after some time, this PR will be closed for inactivity.

If anyone else wishes to continue the work of this PR, please create a PR with a link to this PR for reference.
Thank you for your contributions. 👋

**Comment by sciwhiz12** [comment]
Hello! As per my previous comment, we will be closing this PR for inactivity. If you or another contributor wishes to continue the work of this PR, please create a new PR with a link to this PR for reference. 

Please note that #9017, which is a forward port to 1.19 of this PR, has been merged and therefore should also be linked when making a PR for 1.18.

Thank you for your contributions. 👋

**Pull request closed without merging**